extensions/grpc/runtime/src/main/java/io/quarkus/grpc/runtime/supports/context/GrpcRequestContextGrpcInterceptor.java

```diff
@@ -74,6 +74,30 @@ public void onReady() {
                     }
                 }
 
+                @Override
+                public void onHalfClose() {
+                    boolean activated = activateContext();
+                    try {
+                        super.onHalfClose();
+                    } finally {
+                        if (activated) {
+                            reqContext.deactivate();
+                        }
+                    }
+                }
+
+                @Override
+                public void onCancel() {
+                    boolean activated = activateContext();
+                    try {
+                        super.onHalfClose();
```

Comment:
```suggestion
                        super.onCancel();
```

:-)

Reply:
unfortunately it doesn't work anyway, as I wrote on Zulip...